loader/src/main/kotlin/net/weavemc/loader/bootstrap/Agent.kt

```diff
@@ -41,8 +50,17 @@ fun premain(opt: String?, inst: Instrumentation) {
     Bootstrap.bootstrap(inst)
 }
 
+private fun Instrumentation.tryRetransform(className: String) {
+    val loadedClass = this.allLoadedClasses.firstOrNull { it.name == className }
+    if (loadedClass == null) {
+        Class.forName(className, false, ClassLoader.getSystemClassLoader())
+    } else {
+        this.retransformClasses(loadedClass)
+    }
+}
+
```

Comment:
Maybe I'm missing the point, but, if this is supposed to be an optimization, it probably isn't. A linear search over all loaded classes is most likely way slower than just a lookup (since RuntimeImpl is part of the bootstrap loader, I believe this lookup is guaranteed to be constant time?).

Reply:
I just wanted to prevent a double retransformation since that's what was happening. I was thinking that a Lookup would initialize the class but maybe it's a good idea since it would remove the requirement for .forName

Reply:
Hm, from my testing, any hotspot jvm had already loaded this class (that's the reason I chose to use the retransformation mechanism), therefore the forName would not cause a retransform (this makes sense since the bootstrap loader classes i believe are preinitialized before `main` and `premain`)